native/cocos/core/assets/TextureCube.cpp

```diff
@@ -103,10 +103,74 @@ void TextureCube::setMipmaps(const ccstd::vector<ITextureCubeMipmap> &value) {
     }
 }
 
+void TextureCube::setmipmapAtlas(const TextureCubeMipmapAtlasInfo &value) {
+    if (value.layout.empty()) {
+        return;
+    }
+    _mipmapAtlas = value;
+    const ITextureCubeMipmap &atlas = _mipmapAtlas.atlas;
+    const ccstd::vector<MipmapAtlasLayoutInfo> &layouts = _mipmapAtlas.layout;
+    setMipmapLevel(static_cast<uint32_t>(layouts.size()));
+
+    const MipmapAtlasLayoutInfo &lv0Layout = layouts[0];
+    const ImageAsset *imageAsset = atlas.front;
+
+    reset({lv0Layout.width,
+           lv0Layout.height,
+           imageAsset->getFormat(),
+           static_cast<uint32_t>(layouts.size()),
+           _baseLevel,
+           _maxLevel});
+
+    const uint32_t pixelSize = gfx::GFX_FORMAT_INFOS[static_cast<uint32_t>(imageAsset->getFormat())].size;
+
+    for (size_t level = 0; level < layouts.size(); level++) {
+        const MipmapAtlasLayoutInfo &layoutInfo = layouts[level];
+        uint32_t currentSize = layoutInfo.width * layoutInfo.height * pixelSize;
+
+        //Upload 6 sides by level
+        forEachFace(atlas, [this, currentSize, lv0Layout, layoutInfo, level, pixelSize](ImageAsset *face, TextureCube::FaceIndex faceIndex) {
+            auto *buffer = new uint8_t[currentSize];
+            memset(buffer, 0, currentSize);
+            const uint8_t *data = face->getData();
+            //Splitting Atlas
+            if (level == 0) {
+                memcpy(buffer, data, currentSize);
+            } else {
+                uint32_t bufferOffset = 0;
+                uint32_t dateOffset = lv0Layout.width * lv0Layout.height * pixelSize;
+                uint32_t leftOffset = layoutInfo.left * pixelSize;
+                for (size_t j = 0; j < layoutInfo.height; j++) {
+                    memcpy(buffer + bufferOffset, data + dateOffset + leftOffset, layoutInfo.width * pixelSize);
+                    bufferOffset += layoutInfo.width * pixelSize;
+                    dateOffset += lv0Layout.width * pixelSize;
```

Comment:
Same as ts, it's a waste to copy image data from the atlas, just support upload region in SimpleTexture and TextureCube